### PR TITLE
Add some aria attributes

### PR DESCRIPTION
**Pull request by jvz**
I was reading about [ARIA roles](https://www.w3.org/TR/wai-aria-1.1/#role_definitions) and felt like adding a few obvious ones. The breadcrumb javascript code would be nice to have updated to dynamically change the aria-hidden attributes, but it's not super clear to me how to do that cleanly.

### CODE REVIEW

File: core/src/main/resources/lib/layout/layout.jelly [hunk 1]
```diff
@@ -218,7 +218,7 @@ THE SOFTWARE.
           </j:if>
         </div>
       </j:if>
-      <div id="main-panel">
+      <div id="main-panel" role="main">
```
Comment (fqueiruga) [inline core/src/main/resources/lib/layout/layout.jelly:221]: Is there any `<main>` tag in use? Semantic HTML is better than using aria.

File: core/src/main/resources/lib/layout/layout.jelly [hunk 2]
```diff
@@ -189,7 +189,7 @@ THE SOFTWARE.
     <!-- for accessibility, skip the entire navigation bar and etc and go straight to the head of the content -->
     <a href="#skip2content" class="skiplink">Skip to content</a>
 
-    <div id="page-head">
+    <div id="page-head" role="heading">
```
Comment (fqueiruga) [inline core/src/main/resources/lib/layout/layout.jelly:192]: `<l:pageHeader>` already contains a `<header>` tag if that's what you are looking for. That said, AFAIK the `heading` role is applied to headlines and titles, not blocks. So this change would be invalid either way.
```suggestion
    <div id="page-head">
```

File: core/src/main/resources/lib/layout/breadcrumbBar.jelly [hunk 3]
```diff
@@ -34,42 +34,38 @@ THE SOFTWARE.
...
-                      ${anc.object.displayName}
-                    </a>
+  <st:adjunct includes="lib.layout.breadcrumbs" />
+  <div class="top-sticker noedge">
+    <div class="top-sticker-inner">
+      <div class="breadcrumbs__wrapper">
+        <ul id="breadcrumbs" role="menubar">
+          <j:forEach var="anc" items="${request.ancestors}">
+            <j:if test="${h.isModel(anc.object) and anc.prev.url!=anc.url}">
+              <li class="item" role="menuitem" aria-haspopup="true" aria-expanded="false">
```
Comment (fqueiruga) [inline core/src/main/resources/lib/layout/breadcrumbBar.jelly:44]: `haspopup` and `expanded` should not be used if we will not then update their values based on the dropdown state. They should be removed IMO

File: core/src/main/resources/lib/layout/breadcrumbBar.jelly [hunk 4]
```diff
@@ -34,42 +34,38 @@ THE SOFTWARE.
     use the &lt;l:breadcrumb> tag.
   </st:documentation>
 
-  <tr id="top-nav">
```
Comment (fqueiruga) [inline core/src/main/resources/lib/layout/breadcrumbBar.jelly:37]: Why did you remove this? Did you perform search across the Jenkinsci org to see how this would affect plugins?

Comment (fqueiruga) [reply core/src/main/resources/lib/layout/breadcrumbBar.jelly:37]: Also, if there's a `<tr>`, there must be a wrapping table somewhere.

**Review by fqueiruga — CHANGES_REQUESTED** [review]
The common understanding is that no aria is better than wrong aria. I see some changes that I think are not semantic and also some that I'm not sure they are right. 

I would only approve the changes to the `stopButton.jelly` and `tasks.jelly` and suggest that other files be unchanged by this PR

File: core/src/main/resources/lib/layout/breadcrumbBar.jelly (hunk 4 above)
Comment (jvz) [reply core/src/main/resources/lib/layout/breadcrumbBar.jelly:37]: There used to be until that table-div migration.

Comment (fqueiruga) [reply core/src/main/resources/lib/layout/breadcrumbBar.jelly:37]: It should be unrelated, as table-div migration is scoped to just form elements

File: core/src/main/resources/lib/layout/layout.jelly (hunk 1 above)
Comment (jvz) [reply core/src/main/resources/lib/layout/layout.jelly:221]: Not yet, but that would be a better replacement here.

File: core/src/main/resources/lib/layout/breadcrumbBar.jelly (hunk 3 above)
Comment (jvz) [reply core/src/main/resources/lib/layout/breadcrumbBar.jelly:44]: Yeah, I think I left this here accidentally as I was hoping to dynamically adjust those properties in the corresponding JS, but I don't think I was able to figure out a clean way to introduce that.

File: core/src/main/resources/lib/layout/breadcrumbBar.jelly (hunk 4 above)
Comment (jvz) [reply core/src/main/resources/lib/layout/breadcrumbBar.jelly:37]: Well, I only noticed this because the tr and td weren't showing up in the DOM inspector, and I noticed in the view source tab that those elements were highlighted as syntax errors as it was wrapped in a div. The breadcrumb bar is only invoked from the main layout pages as far as I can tell. I was originally hoping to remove the use of YUI in this widget, but then I got distracted by the ARIA stuff.

Comment (fqueiruga) [reply core/src/main/resources/lib/layout/breadcrumbBar.jelly:37]: It should be a separate PR IMO

File: core/src/main/resources/lib/layout/breadcrumbBar.jelly (hunk 3 above)
Comment (fqueiruga) [reply core/src/main/resources/lib/layout/breadcrumbBar.jelly:44]: Then I suggest removing these attributes

**Review by fqueiruga — CHANGES_REQUESTED** [review]

File: core/src/main/resources/lib/layout/breadcrumbBar.jelly (hunk 4 above)
Comment (zbynek) [reply core/src/main/resources/lib/layout/breadcrumbBar.jelly:37]: There is no wrapping table anymore since 2014 https://github.com/jenkinsci/jenkins/commit/f277502a8304e38f51160897b9cb2e4ed9910e2c so removing the tr/td here seems fine (the browser just ignores those, so any plugins depending on `#top-nav` selector being in DOM would fail anyway). That being said maybe the one CSS reference to `#top-nav` in core can also be removed in this PR.

File: core/src/main/resources/lib/layout/tasks.jelly [hunk 5]
```diff
@@ -34,7 +34,7 @@ THE SOFTWARE.
 -->
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:d="jelly:define">
-	<div id="tasks">
+	<div id="tasks" role="navigation">
```
Comment (zbynek) [inline core/src/main/resources/lib/layout/tasks.jelly:37]: If we prefer semantic tags over roles as suggested in [Félix's comment](https://github.com/jenkinsci/jenkins/pull/4914/files#r477363651), this should be `<nav id="tasks">`.

**Review by zbynek — CHANGES_REQUESTED** [review]
Nice to see some accessibility improvements :) I agree with @fqueiruga that `aria-expanded="false"` should not be used if it's not updated dynamically. I think that's the only blocking issue here.

**Comment by oleg-nenashev** [comment]
@jvz could you please fix the merge conflict?

**Comment by jvz** [comment]
This was mostly an experiment; I don't think I'll have time to go back and update this.
